homeassistant/components/recorder/statistics.py

```diff
@@ -2309,9 +2309,15 @@ def _import_statistics_with_session(
 
     # We just inserted new short term statistics, so we need to update the
     # ShortTermStatisticsRunCache with the latest id for the metadata_id
+    #
+    # Because we are in the same session and we want to read rows
+    # that have not been flushed yet, we need to pass orm_rows=True
+    # to cache_latest_short_term_statistic_id_for_metadata_id
+    # to ensure that it gets the rows that were just inserted
+    #
     run_cache = get_short_term_statistics_run_cache(instance.hass)
     cache_latest_short_term_statistic_id_for_metadata_id(
-        run_cache, session, metadata_id
+        run_cache, session, metadata_id, orm_rows=True
```

Comment:
We need True here because have added data to the current session without a `.flush()`